pkg/eestream/decode.go

```diff
@@ -98,21 +99,25 @@ func (dr *decodedReader) Read(p []byte) (n int, err error) {
 func (dr *decodedReader) Close() error {
 	// cancel the context to terminate reader goroutines
 	dr.cancel()
-	// avoid double close of readers
 	errorThreshold := len(dr.readers) - dr.scheme.RequiredCount()
+	var closeGroup errs2.Group
+	// avoid double close of readers
 	dr.close.Do(func() {
-		var errlist errs.Group
-		// close the readers
 		for _, r := range dr.readers {
-			err := r.Close()
-			if err != nil {
-				errlist.Add(err)
-				errorThreshold--
-			}
+			r := r
```

Comment:
this is such a fun go footgun. might be worth leaving a comment how necessary this line is and what subtle bugs will befall you if you remove it